src/boot/efi/pe.c

```diff
@@ -224,6 +275,73 @@ static void pe_locate_sections(
                 }
 }
 
+static bool looking_for_dbauto(const char *const section_names[]) {
+        assert(section_names);
+
+        for (size_t i = 0; section_names[i]; i++)
+                if (pe_section_name_equal(section_names[i], ".dtbauto"))
+                        return true;
+         return false;
+}
+
+static void pe_locate_sections(
+                const PeSectionHeader section_table[],
+                size_t n_section_table,
+                const char *const section_names[],
+                size_t validate_base,
+                PeSectionVector sections[]) {
+
+        if (!looking_for_dbauto(section_names))
+                return pe_locate_sections_internal(
+                                  section_table,
+                                  n_section_table,
+                                  section_names,
+                                  validate_base,
+                                  /* device_base */ NULL,
+                                  /* device */ NULL,
+                                  sections);
+
+        /* It doesn't make sense not to provide validate_base here */
+        assert(validate_base != 0);
+
+        /* Find HWIDs table and search for the current device */
```

Comment:
the whole block that follows should be enclosed in `if (!firmware_devicetree_exists())`, no? because we don't want to bother with the hwids stuff at all if the firmware exposes a DT, right?

Reply:
i.e. the if block should end right before the 2nd pe_locate_sections_internal() call, so tha tit is called with device == NULL if firmware_devicetree_exists() is true.